src/components/dev-hub/cms-form.js

```diff
@@ -0,0 +1,90 @@
+import React, { useReducer } from 'react';
+import Checkbox from '@leafygreen-ui/checkbox';
+import Input from './input';
+import TextArea from './text-area';
+
+const getInitialFormState = () => ({});
+
+const FORM_ELEMENT_TYPES = {
+    CHECKBOXES: 'Checkboxes',
+    EMAILINPUT: 'EmailInput',
+    TEXTAREA: 'Textarea',
+};
+
+const handleCMSFormChange = (state, { field, value }) => ({
+    ...state,
+    [field]: value,
+});
+
+const mapTypeToFormElement = (type, labels, state, dispatch, name) => {
+    switch (type) {
+        case FORM_ELEMENT_TYPES.CHECKBOXES:
+            return labels.map(({ label }, index) => (
+                <Checkbox
+                    key={`${name}-${index}`}
+                    bold
+                    onChange={e =>
+                        dispatch({
+                            field: `${name}-${index}`,
+                            value: e.target.value,
+                        })
+                    }
+                    darkMode={true}
+                    label={label}
+                />
+            ));
+        case FORM_ELEMENT_TYPES.EMAILINPUT:
+            return (
+                <Input
+                    key={labels[0].label}
+                    bold
+                    darkMode={true}
+                    label={labels[0].label}
```

Comment:
Nice work! Just have a suggestion, if it is possible could we change our data on server and create object 
{ label: 'Some Labe', placeholder: 'SomePlaceholder', ...another_props } instead of {labels: [{label}, {placeholder}]}?

Reply:
Good question! I was able to add a new field for `placeholder` but cannot remove the object within the array for now. The Strapi CMS out of the box limits what types we can use in an array to be an object. This can be changed but for now I would like to leave as object since we can change in the future.